Not able to compile

Unfortunately compiling with dallasng component always ends in error messages:

"
In file included from src/esphome/components/sensor/sensor.h:3,
                 from src/esphome/components/dallasng/dallasng_component.h:4,
                 from src/esphome/components/dallasng/dallasng_component.cpp:1:
src/esphome/components/dallasng/dallasng_temperature_sensor.h: In member function 'bool esphome::dallasng::DallasNgTemperatureSensor::try_get_temperature_c(float*)':
src/esphome/components/dallasng/dallasng_temperature_sensor.h:119:31: error: format '%d' expects argument of type 'int', but argument 6 has type 'long int' [-Werror=format=]
                 ESP_LOGD(TAG, "'%s' got temperature as %d (%f)", get_name().c_str(), temp, *value);
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/core/log.h:69:36: note: in definition of macro 'ESPHOME_LOG_FORMAT'
 #define ESPHOME_LOG_FORMAT(format) format
                                    ^~~~~~
src/esphome/core/log.h:152:28: note: in expansion of macro 'esph_log_d'
 #define ESP_LOGD(tag, ...) esph_log_d(tag, __VA_ARGS__)
                            ^~~~~~~~~~
src/esphome/components/dallasng/dallasng_temperature_sensor.h:119:17: note: in expansion of macro 'ESP_LOGD'
                 ESP_LOGD(TAG, "'%s' got temperature as %d (%f)", get_name().c_str(), temp, *value);
                 ^~~~~~~~
src/esphome/components/dallasng/dallasng_component.cpp: In member function 'virtual void esphome::dallasng::DallasNgComponent::setup()':
src/esphome/components/dallasng/dallasng_component.cpp:13:18: error: 'class OneWireNg' has no member named 'searchReset'
       one_wire_->searchReset();
                  ^~~~~~~~~~~
src/esphome/components/dallasng/dallasng_component.cpp:16:18: error: 'class OneWireNg' has no member named 'searchReset'
       one_wire_->searchReset();
                  ^~~~~~~~~~~
src/esphome/components/dallasng/dallasng_component.cpp:17:25: error: 'class OneWireNg' has no member named 'search'
       while (one_wire_->search(id, false) == OneWireNg::EC_MORE)
                         ^~~~~~
cc1plus.exe: some warnings being treated as errors
*** [.pioenvs\temperature\src\esphome\components\dallasng\dallasng_component.o] Error 1
"

My used yaml code is:

`substitutions:
  name: temperature

esphome:
  name: ${name}
  
esp32:
  board: wemos_d1_mini32
  framework:
    type: esp-idf
    version: recommended

external_components:
  - source: github://nrandell/dallasng

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

mqtt:
  broker: !secret mqtt_host
  port: !secret mqtt_port
  username: !secret mqtt_username
  password: !secret mqtt_password
  id: mqtt_client

logger:
  level: DEBUG

dallasng:
  - pin: GPIO13 #GPIO13
    update_interval: 5s

sensor:
  - platform: dallasng
    index: 0
    name: Temperature
    resolution: 12
    #expire_after: 1000s
    filters:
      - filter_out: NAN`
  
ESPHome-Version is: 2022.9.4

--- a/components/dallasng/__init__.py
+++ b/components/dallasng/__init__.py
@@ -20,6 +20,7 @@ CONFIG_SCHEMA = cv.Schema(
 async def to_code(config):
     var = cg.new_Pvariable(config[CONF_ID])
     await cg.register_component(var, config)
+    cg.add_build_flag("-DOWNG_NO_NATIVE_CONFIG")
     cg.add_library("https://github.com/pstolarz/OneWireNg", None)
 
     pin = await cg.gpio_pin_expression(config[CONF_PIN])

--- a/components/dallasng/dallasng_temperature_sensor.h
+++ b/components/dallasng/dallasng_temperature_sensor.h
@@ -116,7 +116,7 @@ namespace esphome
                 DSTherm::Scratchpad &s = scratchpad;
                 long temp = s.getTemp();
                 *value = (float)temp / 1000.0f;
-                ESP_LOGD(TAG, "'%s' got temperature as %d (%f)", get_name().c_str(), temp, *value);
+                ESP_LOGD(TAG, "'%s' got temperature as %ld (%f)", get_name().c_str(), temp, *value);
                 return true;
             }
 

